.github/workflows/functionality-tests.yml

```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         node_version:
-          - 16
```

Comment:
we're still using `actions/setup-node@v1` which is quite old. by upgrading to `v3` we can make use of the `node-version-file` so that it uses the same version declared in the `.tool-versions` file
https://github.com/actions/setup-node#usage